src/pages/index.js

```diff
@@ -4,53 +4,136 @@ import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
 import Section from "../components/Section.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
+import Api from "../components/Api.js";
 import "./index.css";
 import jacquesCousteau from "../images/jacques-cousteau.jpg";
 import logoImage from "../images/logo.svg";
 import {
   settings,
-  initialCards,
   profileEditButton,
   cardAddButton,
+  avatarEditButton,
 } from "../utils/constants.js";
 
 //Functions
 function handleEditProfileFormSubmit(data) {
-  userInfo.setUserInfo(data);
-  profilePopupForm.close();
+  profilePopupForm.setLoading(true);
+  api
+    .editProfile({ name: data.name, description: data.description })
+    .then((res) => {
+      userInfo.setUserInfo({ name: res.name, description: res.about });
+      profilePopupForm.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+  profilePopupForm.setLoading(false);
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.


That's why the loading effect is not visible



**Please, correct it everywhere in the code**

Reply:
I've made sure each submit has a finally block. As a side note, it still wasn't loading correctly. The problem seemed to be solved when I turned the finally block into an anonymous function since so it could receive the argument and then still run a different function. Would that be correct? It wouldn't work any other way.

Reply:
> an anonymous function

Right